app/controllers/concerns/authentication_and_sso_concerns.rb

```diff
@@ -105,6 +105,16 @@ def set_session_expiration_header
     headers['X-Session-Expiration'] = @session_object.ttl_in_time.httpdate if @session_object.present?
   end
 
+  def log_sso_info
+    action = request.controller_instance.controller_path.classify.to_s
+    action += 'Controller#'
+    action += request.parameters['action'].to_s
```

Comment:
You can get the full name of the controller class including `Controller` with `self.class.to_s`.

The action is available in the controller with `action_name`

So you should be able to clean this up a bit with something like this:

```ruby
action = "#{self.class.to_s}##{action_name}"
```

Reply:
👍 I'll update